crates/interpreter/src/instructions/opcode.rs

```diff
@@ -1,453 +1,313 @@
-use crate::gas;
-use crate::primitives::SpecId;
+//! EVM opcode definitions and utilities.
 
-#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+use super::*;
+use crate::{
+    gas,
+    primitives::{Spec, SpecId},
+    Host, Interpreter,
+};
+use core::fmt;
+
+macro_rules! opcodes {
```

Comment:
This macro declares all of:
- `const NAME: u8 = $value;`
- `OPCODE_JUMPMAP[$value] = Some(stringify!($name));`
- `eval` dispatch

Left out gas, which is improved below.

Reply:
was wondering if `gas` should be part of the opcode, but maybe not? it'd be nice if they were all declared in one place, but ig fine as-is?